crates/polars-plan/src/dsl/function_expr/range/utils.rs

```diff
@@ -19,3 +22,86 @@ pub(super) fn ensure_range_bounds_contain_exactly_one_value(
     );
     Ok(())
 }
+
+/// Create a ranges column from the given start/end columns and a range function.
+pub(super) fn ranges_impl_broadcast<T, U, F>(
+    start: &ChunkedArray<T>,
+    end: &ChunkedArray<T>,
+    range_impl: F,
+    builder: &mut ListPrimitiveChunkedBuilder<U>,
+) -> PolarsResult<Series>
+where
+    T: PolarsIntegerType,
+    U: PolarsIntegerType,
+    F: Fn(T::Native, T::Native, &mut ListPrimitiveChunkedBuilder<U>) -> PolarsResult<()>,
+{
+    match (start.len(), end.len()) {
+        (len_start, len_end) if len_start == len_end => {
+            build_ranges::<T, U, F>(start.into_iter(), end.into_iter(), range_impl, builder)?;
+        },
+        (1, len_end) => {
+            let start_scalar = start.get(0);
+            match start_scalar {
+                Some(start) => build_ranges::<T, U, F>(
+                    repeat(Some(start)),
```

Comment:
nit: can we use the full name here. `std::iter::repeat` is more familiar to people than `repeat` (the latter I start wondering where it is implemented).